staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3971,6 +3977,15 @@ type LoadBalancerIngress struct {
 	// (typically AWS load-balancers)
 	// +optional
 	Hostname string `json:"hostname,omitempty" protobuf:"bytes,2,opt,name=hostname"`
+
+	//The list has one entry per service port in the manifest
+	// +listType=map
```

Comment:
Why not atomic?  Do we really need to patch individual records?  The key for the map is multi-field, which is unpleasant